#3 Initialize telegram handler

savvasal opened the pull request.

gmponos reviewed (commented):
Hi.. sorry for any delay.. was trying to setup the ci.. can you merge with master and check if CI passes..?

savvasal committed:
'#3 Implement telegram handler'

Try a different phpcs

savvasal committed:
Run cs fixer

savvasal committed:
Fix phpstan

gmponos commented on src/TelegramHandler.php:
```diff
@@ -8,26 +8,69 @@
+     * @param int|string $chatId
+     * @param ClientInterface $client
+     * @param RequestFactoryInterface $requestFactory
+     * @param int $level
```
Level can be a string as well.

gmponos commented on tests/TelegramHandlerTest.php:
```diff
@@ -0,0 +1,59 @@
+            ->method('write')
+            ->with('{"chat_id":1234,"text":"This is an error message"}');
+        $mockBody->expects($this->once())->method('rewind');
+        $mockRequest = $this->getMockBuilder(RequestInterface::class)->getMock();
```
There is a helper function from phpstorm `createMock` let's use that one.

gmponos commented on tests/TelegramHandlerTest.php:
```diff
@@ -0,0 +1,59 @@
+<?php
+
```
declare strict types

gmponos commented on src/TelegramHandler.php:
```diff
@@ -8,26 +8,69 @@
+            'text' => $record['formatted'],
+        ];
+        $jsonBody = \json_encode($body);
+        if (false === $jsonBody) {
```
it might be better to check `json_last_error()` after the check add an inline docblock in order to explain to phpstan that jsonBody is a string.

gmponos commented on src/TelegramHandler.php:
```diff
@@ -8,26 +8,69 @@
+     * @see https://core.telegram.org/bots/api#sendmessage
+     *
+     * @param array $record
+     * @return RequestInterface
```
Let's remove superflous docblocks.

There is also a php-cs-fixer rule: https://mlocati.github.io/php-cs-fixer-configurator/#version:2.16|fixer:no_superfluous_phpdoc_tags

maybe we should include this rule.. can you add it?

savvasal committed:
Improvements
- Use createMock
- Use json_last_error
- Add no_superfluous_phpdoc_tags cs rule
